### PR TITLE
[docs] Document deprecated functions from #20142 in the maintainer guide.

### PR DESCRIPTION
**Pull request by BillyONeal**
See discussion at https://github.com/microsoft/vcpkg/pull/20142#issuecomment-921146572

### CODE REVIEW

**Review by PhoebeHui — APPROVED** [review]

File: docs/maintainers/maintainer-guide.md [hunk 1]
```diff
@@ -46,6 +46,25 @@ At this time, the following helpers are deprecated:
 2. `vcpkg_apply_patches()` should be replaced by the `PATCHES` arguments to the "extract" helpers (e.g. [`vcpkg_from_github()`](vcpkg_from_github.md))
 3. `vcpkg_build_msbuild()` should be replaced by [`vcpkg_install_msbuild()`](vcpkg_install_msbuild.md)
 4. `vcpkg_copy_tool_dependencies()` should be replaced by [`vcpkg_copy_tools()`](vcpkg_copy_tools.md)
+5. `vcpkg_install_cmake` should be replaced by [`vcpkg_cmake_install()`](vcpkg_cmake_install.md) (from port `vcpkg-cmake`)
+6. `vcpkg_build_cmake` should be replaced by [`vcpkg_cmake_build()`](vcpkg_cmake_build.md) (from port `vcpkg-cmake`)
+7. `vcpkg_configure_cmake` should be replaced by [`vcpkg_cmake_configure()`](vcpkg_cmake_configure.md) after removing `PREFER_NINJA` (from port `vcpkg-cmake`)
```
Comment (dg0yt) [inline docs/maintainers/maintainer-guide.md:51]: I would prefer to see `vcpkg_cmake_configure` as the first added point(i.e. "5."), to match the natural order in portfiles.

Comment (BillyONeal) [reply docs/maintainers/maintainer-guide.md]: Done!

**Review by PhoebeHui — APPROVED** [review]

**Comment by Osyotr** [comment]
Would you kindly also update vcpkg-tool? It suggests to use deprecated `vcpkg_fixup_cmake_targets`
https://github.com/microsoft/vcpkg-tool/blob/733e90d1bc3f4f8188d57d73e2d20765c232cc26/src/vcpkg/postbuildlint.cpp#L265

**Comment by BillyONeal** [comment]
> Would you kindly also update vcpkg-tool? It suggests to use deprecated `vcpkg_fixup_cmake_targets`
> https://github.com/microsoft/vcpkg-tool/blob/733e90d1bc3f4f8188d57d73e2d20765c232cc26/src/vcpkg/postbuildlint.cpp#L265

Done: https://github.com/microsoft/vcpkg-tool/pull/198

**Pull request merged**